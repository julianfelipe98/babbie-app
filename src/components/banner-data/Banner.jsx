import React from "react";
import "./Banner.css";
import Button from '@material-ui/core/Button';
import { makeStyles } from "@material-ui/core";

const useStyles = makeStyles({
    root: {
      background: 'linear-gradient(45deg, #FE6B8B 30%, #FF8E53 90%)',
      border: 0,
      borderRadius: 50,
      boxShadow: '0 3px 5px 2px rgba(255, 105, 135, .3)',
      color: 'white',
      height: 48,
      padding: '0 60px',
    },
  });
export const Banner = () => {
    const classes = useStyles();
    return (
        <div id="banner">
            <div id="text-banner">
                <p>Aplicaion para la gestion veterinaria</p>
                <h1>Â¡El amor es una palabra de cuatro patas!</h1>
                <p>Descarga nuestra app en dispositivo android</p>
            </div>
            <Button className={classes.root}>Descargar</Button>
        </div>
    );
};
