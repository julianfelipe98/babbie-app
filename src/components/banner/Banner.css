#banner1 {
  background-image: url("../../assets/banner.svg");
  background-size: cover;
  width: 100%;
  height: 70vh;
  color: aliceblue;
  text-align: center;
}
#banner2 {
  background-image: url("../../assets/banner2.svg");
  background-size: cover;
  width: 100%;
  height: 70vh;
  color: aliceblue;
  text-align: center;
  margin-top: 5rem;
}

#text-banner {
  font-family: "Trebuchet MS", "Lucida Sans Unicode", "Lucida Grande",
    "Lucida Sans", Arial, sans-serif;
  padding-top: 20vh;
}
#download-btn {
  background-color: #497df9;
  border: none;
  border-radius: 8px;
  color: #fff;
  font-size: 13px;
  font-weight: bold;
  padding: 20px 35px;
}
#download-btn2 {
  background-color: #497df9;
  border: none;
  border-radius: 25px;
  color: #fff;
  font-size: 18px;
  font-weight: bold;
  padding: 15px 35px;
}
