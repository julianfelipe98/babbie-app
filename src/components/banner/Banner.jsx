import React from "react";
import "./Banner.css";

export const Banner = ({ id, title, subtitle, paragraph,styleBtn,btn }) => {
  return (
    <div id={`banner${id}`}>
      <div id="text-banner">
        <p id="title">{title}</p>
        <h1 id="subtitle">{subtitle}</h1>
        <p id="paragraph">{paragraph}</p>
      </div>
      <button id={styleBtn}>{btn}</button>
    </div>
  );
};
