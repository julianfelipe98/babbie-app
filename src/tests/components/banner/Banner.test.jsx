import { shallow } from "enzyme";
import { Banner } from "../../../components/banner/Banner";

describe("Banner component tests", () => {
  const title = "title";
  const subtitle = "subtitle";
  const paragraph = "paragraph";
  let wrapper ;
  beforeEach(() => {
    wrapper = shallow(
      <Banner title={title} subtitle={subtitle} paragraph={paragraph} />
    );
  });
  test("should show the correct title by props", () => {
    const textTitle = wrapper.find("#title").text();
    expect(textTitle).toBe(title);
  });
  test("should show the correct subtitle by props", () => {
    const textSubtitle = wrapper.find("#subtitle").text();
    expect(textSubtitle).toBe(subtitle);
  });
  test("should show the correct paragraph by props", () => {
    const textParagraph = wrapper.find("#paragraph").text();
    expect(textParagraph).toBe(paragraph);
  });
  test("should render <BannerApp/> in the correct way", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
