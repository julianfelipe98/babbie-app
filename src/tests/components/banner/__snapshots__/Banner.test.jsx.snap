// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner component tests should render <BannerApp/> in the correct way 1`] = `
<div
  id="bannerundefined"
>
  <div
    id="text-banner"
  >
    <p
      id="title"
    >
      title
    </p>
    <h1
      id="subtitle"
    >
      subtitle
    </h1>
    <p
      id="paragraph"
    >
      paragraph
    </p>
  </div>
  <button />
</div>
`;
